docs/spec.md

```diff
@@ -265,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
 
-In order to distinguish between array literals and array indices, there
-cannot be any whitespace between array variable and index.
+```evy
+arr := ["abc" 123] // []any
+print "Type of arr:" (typeof arr)
+```
 
-    arr := ["a" "b"]
-    print arr[1]  // index: b
-    print arr [1] // literal: [a b] [1]
-    arr[0] = "A"
-    arr [1] = "B" // invalid
+outputs
 
-See section [whitespace](#whitespace) for further details.
+```evy:output
+Type of arr: []any
+```
+
+The function `len arr` returns the length of the array, which is the
+number of elements in the array. The loop `for el := range arr`
+iterates over all elements of the array in order. Arrays can be
+concatenated with the `+` operator, for example `arr2 := arr + arr`.
+
+The elements of an array can be accessed via index starting at `0`. In
+the example `arr := ["abc" 123]` the first element in the array
+`arr[0]` is `"abc"`.
+
+The empty array becomes `[]any` in inferred declarations, otherwise the
+empty array literal assumes the array type required by the assigned
+variable or parameter. `arr:[]any` and `arr := []` are equivalent.
+
+In order to distinguish between array literals and array indices, there
+cannot be any whitespace between array variable and index. For example,
+the following code
+
+```evy
+arr := ["a" "b"]
+print 1 arr[1] // index
+print 2 arr [1] // literal
+arr[0] = "A"
+print 3 arr
+// arr [1] = "B" // whitespace before `[` is invalid
+```
+
+outputs
+
+```evy:output
+1 b
+2 [a b] [1]
+3 [A b]
+```
 
 ## Maps
 
 Maps are key-value stores, where the values can be looked up by their
-key
+key, for example `m := { key1:"value1" key2:"value2" }`.
+
+Map values can be accessed with the dot expression, for example
+`map.key`. If maps are accessed via the dot expression the key must
+match the grammars `ident` production. Map values can also be accessed
+with an index expression which allows for evaluation, non-ident keys
+and variable usage. For example the following code
+
+```evy
+m := {letters:"abc"}
+print 1 m.letters
+print 2 m["letters"]
+
+key := "German letters"
+m[key] = "äöü"
+print 3 m[key]
+print 4 m["German letters"]
+```
+
+outputs
+
+```evy:output
+1 abc
+2 abc
+3 äöü
+4 äöü
+```
 
-    m := { key1:"value1" key2:"value2" }
+The `has` function tests for the existence of a key in a map. The
+following code
 
-Map keys must be strings that match the grammars `ident` production. Map
-values can be accessed with the dot expression, for example `map.key`.
-Map values can also be accessed with an index which allows for
-evaluation and variable usage:
+```evy
+m := {letters:"abc"}
+print 1 (has m "letters")
+print 2 (has m "digits")
+```
 
-    m := { letters: "abc" }
-    print m.letters    // abc
-    print m["letters"] // abc
+outputs
 
-    s := "letters"
-    print m[s]         // abc
+```evy:output
+1 true
+2 false
+```
 
-The `has` function tests for the existence of a key in a map:
+The `del` function deletes a key from a map if it exists. The following
```

Comment:
...exists and does nothing if the key does not exist.

Reply:
yes!